tutorials/chapter2.md

```diff
@@ -263,4 +263,58 @@ int main(int argc, char **argv) {
 
 ### DDS
 
+When Klepsydra is compiled and installed with DDS support, three DDS specific libraries are installed: `kpsr_dds_serialization_datamodel`, `kpsr_dds_core` and `kpsr_dds_core_datamodel`, and are available to be linked against using the environment variable `KLEPSYDRA_DDS_LIBRARIES`. The `kpsr_dds_serialization` provides DDS compatible types for primitive data types. As DDS requires message data types to be specified in an *.idl file, Klepsydra provides the primitive data types in wrapped formats which also include the sequence number field. The `kpsr_dds_serialization` also provides mapping functions to conveniently transform data from the primitive type to the dds specific type. 
+
+#### Sending / receiving data - the HelloWorld example:
+
+The HelloWorld example for DDS can then use the same SimplePublisher class as shown for ROS and the complete example would be as shown below:
+
+```cpp
+
+int main(int argc, char **argv) {
+    dds::domain::DomainParticipant dp(0);
+    dds::pub::Publisher publisher(dp);
+    dds::sub::Subscriber subscriber(dp);
+
+    dds::topic::Topic<kpsr_dds_serialization::StringData> stringDataTopic(dp, "stringData");
+    dds::pub::DataWriter<kpsr_dds_serialization::StringData> stringDataWriter(publisher, stringDataTopic);
+    dds::sub::DataReader<kpsr_dds_serialization::StringData> stringDataReader(subscriber, stringDataTopic);
+
+    kpsr::dds_mdlw::ToDDSMiddlewareProvider provider(nullptr);
+    kpsr::dds_mdlw::FromDDSMiddlewareProvider ddsProvider;
+
+    kpsr::Publisher<std::string> * stringDataPublisher =
+            provider.getToMiddlewareChannel<std::string, kpsr_dds_serialization::StringData>("stringData", 0, nullptr, &stringDataWriter);
+
+    SimplePublisher publisher(stringDataPublisher);
+
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> stringDataSafeQueueProvider(nullptr);
+    stringDataSafeQueueProvider.start();
+    ddsProvider.registerToTopic("stringData", &stringDataReader, true, stringDataSafeQueueProvider.getPublisher<std::string>("string", 0, nullptr, nullptr));
+
+    stringDataSafeQueueProvider.getSubscriber<std::string>("string")->registerListener("example", [](std::string &msg) {
+        std::cout << "Received message: " << msg << std::endl; });
+
+    simplePublisher.run();
+
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    stringDataSafeQueueProvider.getSubscriber<std::string>("string")->removeListener("example");
+    stringDataSafeQueueProvider.stop();
+    ddsProvider.unregisterFromTopic("stringData", &stringDataReader);
+}
+```
+
+* The first 6 lines show the standard method for setting up a DDS domain participant with its publisher, subscriber, topic, writer and  reader.
```

Comment:
The text is great, just one thing, let's use the three inverted quotes to mark the class names as code and put the full name with namespace, i.e., ```kpsr::dds_mdlw::DDSEnv```.

Reply:
Fixed